docs/website/docs/guides/developer-tips.md

```diff
@@ -0,0 +1,424 @@
+        can be disassembled using
+        [`llvm-dis`](https://llvm.org/docs/CommandGuide/llvm-dis.html):
+
+        ```console
+        // Build `llvm-dis` from source as needed:
+        $ cmake --build iree-build/ --target llvm-dis
+        $ iree-build/llvm-project/bin/llvm-dis --help
+
+        $ cd /tmp/iree/simple_abs/
+        $ llvm-dis module_abs_dispatch_0_cuda_nvptx_fb.codegen.bc
+        $ cat module_abs_dispatch_0_cuda_nvptx_fb.codegen.ll
+
+        ; ModuleID = 'module_abs_dispatch_0_cuda_nvptx_fb.codegen.bc'
+        source_filename = "abs_dispatch_0"
+
+        declare ptr @malloc(i64)
+
+        declare void @free(ptr)
+
+        declare float @__nv_fabsf(float)
+
+        define void @abs_dispatch_0_generic(ptr noalias readonly align 16 %0, ptr noalias align 16 %1) {
+          %3 = ptrtoint ptr %0 to i64
+          %4 = and i64 %3, 63
+          %5 = icmp eq i64 %4, 0
+          call void @llvm.assume(i1 %5)
+          %6 = ptrtoint ptr %1 to i64
+          %7 = and i64 %6, 63
+          %8 = icmp eq i64 %7, 0
+          call void @llvm.assume(i1 %8)
+          %9 = load float, ptr %0, align 4
+          %10 = call float @__nv_fabsf(float %9)
+          store float %10, ptr %1, align 4
+          ret void
+        }
+
+        !nvvm.annotations = !{!0, !1, !2, !3}
+
+        !0 = !{ptr @abs_dispatch_0_generic, !"kernel", i32 1}
+        !1 = !{ptr @abs_dispatch_0_generic, !"maxntidx", i32 1}
+        !2 = !{ptr @abs_dispatch_0_generic, !"maxntidy", i32 1}
+        !3 = !{ptr @abs_dispatch_0_generic, !"maxntidz", i32 1}
+        ```
+
+<!-- TODO(scotttodd): Link to a playground Colab notebook that dumps files? -->
+
+## Compiling phase by phase
+
+IREE compiles programs through a series of broad phases:
+
+``` mermaid
+graph LR
+  accTitle: Compilation phases overview
+  accDescr: Input to ABI to Flow to Stream to HAL to VM
+
+  A([Input])
+  A --> B([ABI])
+  B --> C([Flow])
+  C --> D([Stream])
+  D --> E([HAL])
+  E --> F([VM])
+```
+
+You can output a program snapshot at intermediate phases with the
+`--compile-to=<phase name>` flag:
+
+```console
+$ cat simple_abs.mlir
+
+func.func @abs(%input : tensor<f32>) -> (tensor<f32>) {
+  %result = math.absf %input : tensor<f32>
+  return %result : tensor<f32>
+}
+
+$ iree-compile simple_abs.mlir --compile-to=abi
+
+module {
+  func.func @abs(%arg0: !hal.buffer_view) -> !hal.buffer_view attributes {iree.abi.stub} {
+    %0 = hal.tensor.import %arg0 "input 0" : !hal.buffer_view -> tensor<f32>
+    %1 = math.absf %0 : tensor<f32>
+    %2 = hal.tensor.export %1 "output 0" : tensor<f32> -> !hal.buffer_view
+    return %2 : !hal.buffer_view
+  }
+}
+```
+
+This is similar to the `--mlir-print-ir-after=` flag, but at clearly defined
+pipeline phases.
+
+Similarly, compilation can be continued from any intermediate phase. This allows
+for interative workflows - compile to a phase, make edits to the `.mlir` file,
+then resume compilation and continue through the pipeline.
```

Comment:
This is a great piece of advice! Could we also show the command to resume compilation from some `.mlir`'?

Reply:
Sure thing. Added a section. Here's how it renders: 
![image](https://github.com/openxla/iree/assets/4010439/591f686e-f143-444b-956e-0cb4b5f1d6f2)

(is changing `absf` to `exp` a useful/clear enough example?)

Reply:
BTW, these test files show other ways of using those features (but I don't really feel like linking them here... could be persuaded):

* https://github.com/openxla/iree/blob/main/tools/test/compile_to_phase.mlir
* https://github.com/openxla/iree/blob/main/tools/test/compile_to_continuation.mlir

Also see

* https://github.com/openxla/iree/blob/main/tools/test/compile_pipelines.mlir